edx_rest_framework_extensions/auth/jwt/middleware.py

```diff
@@ -72,19 +86,87 @@ def _add_missing_jwt_permission_classes(self, view_class):
             view_class.permission_classes += tuple(classes_to_add)
 
     def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: disable=unused-argument
-        # Views as functions store the view's class in the 'view_class' attribute.
-        # Viewsets store the view's class in the 'cls' attribute.
-        view_class = getattr(
-            view_func,
-            'view_class',
-            getattr(view_func, 'cls', view_func),
-        )
+        view_class = _get_view_class(view_func)
 
         view_authentication_classes = getattr(view_class, 'authentication_classes', tuple())
-        if self._includes_base_class(view_authentication_classes, BaseJSONWebTokenAuthentication):
+        if _includes_base_class(view_authentication_classes, BaseJSONWebTokenAuthentication):
             self._add_missing_jwt_permission_classes(view_class)
 
 
+class JwtAuthWithLoginRequiredMiddleware(MiddlewareMixin):
+    """
+    Middleware enables DRF JwtAuthentication authentication class for endpoints
+    using the LoginRequired permission class.
+
+    Enables a non-standard use of DRF, because DRF only supports returning
+    a 401 for unauthorized users, but this middleware allows a DRF endpoint
+    to redirect to login.
+
+    This can be used to convert a plan Django view using @login_required into a
+    DRF APIView, which is useful to enable our DRF JwtAuthentication class.
+
+    Usage Notes:
+    - This middleware must be added before JwtAuthCookieMiddleware.
+    - Only affects endpoints using the LoginRequired permission class.
+
+    See https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0009-jwt-in-session-cookie.rst  # noqa
+    """
+    _LOGIN_REQUIRED_FOUND_CACHE_KEY = 'login_required_found'
+
+    def _get_request_cache(cls):
+        cache_namespace = 'JwtAuthWithLoginRequiredMiddleware'
```

Comment:
Nit: define this as a constant, along with `_LOGIN_REQUIRED_FOUND_CACHE_KEY`.

Reply:
It originally had that, but it made the code messier for a constant that wasn't shared.  I wish it was fine in Python to use the following in a method:
```
CACHE_NAMESPACE = 'JwtAuthWithLoginRequiredMiddleware'
```
Of the two imperfect options: 1) all caps at the class level, or 2) lower case variable, I still prefer what I have.

Reply:
I suggested it, mostly for readability, so that the cache-namespace string is co-located with the cache-key string.

Reply:
Done.  I moved it lower, so less of an issue.